modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java

```diff
@@ -100,14 +112,57 @@ private <R> CompletableFuture<R> sendToReplica(ClusterNode node, ReplicaRequest
 
                 if (response instanceof ErrorReplicaResponse) {
                     var errResp = (ErrorReplicaResponse) response;
-                    res.completeExceptionally(errResp.throwable());
+
+                    if (errResp.throwable() instanceof ReplicaUnavailableException) {
+                        pendingInvokes.compute(node, (clusterNode, fut) -> {
+                            AwaitReplicaRequest awaitReplicaReq = REPLICA_MESSAGES_FACTORY.awaitReplicaRequest()
```

Comment:
Seems that we may put message preparation inside fut == null clause.

Reply:
fixed